### PR TITLE
Implement SAFE_SSCANF macro and underlying function

### PR DESCRIPTION
**Pull request by ilmanzo**
Added also a very simple unit test to exercise it

### CODE REVIEW

File: lib/tst_safe_macros.c [hunk 1]
```diff
@@ -625,3 +625,14 @@ void safe_print_file(const char *file, const int lineno, char *path)
...
 	safe_fclose(file, lineno, NULL, pfile);
 }
+
+int safe_sscanf(const char *file, const int lineno, const char *restrict buffer, const char *restrict format, ...)
+{
+	va_list args;
+	va_start(args, format);
+	int ret = vsscanf(buffer, format, args);
+	va_end(args);
+	if (ret == EOF)
```
Comment (metan-ucw) [inline lib/tst_safe_macros.c:635]: This does not actually fail if we have less than expected number of conversion but some successful. You have to count the conversions and make sure that we got exactly the right number. We do have `tst_count_scanf_conversions()` function for exactly that.

Comment (ilmanzo) [reply lib/tst_safe_macros.c:635]: Thanks for the hint; from first instructions it wasn't clear you wanted to have that kind of check in place. Makes sense 👍🏻, will do ASAP

File: lib/tst_safe_macros.c [hunk 2]
```diff
@@ -625,3 +625,17 @@ void safe_print_file(const char *file, const int lineno, char *path)
...
 	safe_fclose(file, lineno, NULL, pfile);
 }
+
+int safe_sscanf(const char *file, const int lineno, const char *restrict buffer, const char *restrict format, ...)
+{
+	va_list args;
+	va_start(args, format);
+	int ret = vsscanf(buffer, format, args);
+	va_end(args);
+	int placeholders=tst_count_scanf_conversions(format);
```
Comment (pevik) [inline lib/tst_safe_macros.c:635]: nit: Adding a blank line here makes sources more readable (separate variable declaration from the rest.
Also please add a space around `=`.

File: lib/tst_safe_macros.c [hunk 3]
```diff
@@ -625,3 +625,17 @@ void safe_print_file(const char *file, const int lineno, char *path)
...
+	if (ret != placeholders)
```
Comment (pevik) [inline lib/tst_safe_macros.c:636]: nit: blank line before `if ()` and `return` also helps readability (also makes sources a bit consistent - it's used in whole file).

File: lib/tst_safe_macros.c (hunk 1 above)
Comment (pevik) [reply lib/tst_safe_macros.c:638]: I would first handle `EOF` as the other check is IMHO more specific.

Also, does it make sense to print `errno` (with `TERRNO`)? Man states:

       EILSEQ Input byte sequence does not form a valid character.

       EINVAL Not enough arguments; or format is NULL.

       ENOMEM Out of memory.

**Comment by pevik** [comment]
Final thing, please fix formatting:

```
$ cd lib; make check-tst_safe_macros
...
CHECK lib/tst_safe_macros.c
tst_safe_macros.c:629: WARNING: Missing a blank line after declarations
tst_safe_macros.c:631: WARNING: Missing a blank line after declarations
tst_safe_macros.c:639: ERROR: trailing whitespace
```

(note: the last `LTP-003: Symbol 'safe_prot_to_str'` is old issue with fix in our mailing list)

With that we can IMHO merge it.
@Cyril: any hint or can I merge with your RBT?

```
Reviewed-by: Petr Vorel <pvorel@suse.cz>
```

**Pull request closed without merging**